db/channel_cache.go

```diff
@@ -343,8 +410,17 @@ func (c *channelCache) insertChange(log *LogEntries, change *LogEntry) {
 
 // Prepends an array of entries to this one, skipping ones that I already have.
 // The new array needs to overlap with my current log, i.e. must contain the same sequence as
-// c.logs[0], otherwise nothing will be added because the method can't confirm that there are no
-// missing sequences in between.
+// the oldest entry in the cache (c.logs[0]), otherwise nothing will be added because the
+// method can't confirm that there are no missing sequences in between.
+//
+// changesValidFrom represents the validfrom value of the query, and should always be LTE the
+// oldest value (lowest sequence number) in the cache, although this is not strictly enforced.
```

Comment:
There's no requirement for it to be LTE the oldest value in the cache, but it should be LTE the oldest value in the set of changes being prepended.

Reply:
Momentary lapse of reason!  Thanks for catching